tasks/broad/RNAWithUMIsTasks.wdl

```diff
@@ -614,3 +614,130 @@ task MarkDuplicatesUMIAware {
     }
 }
 
+
+task formatPipelineOutputs {
+  input {
+    String output_basename
+    String transcriptome_bam
+    String transcriptome_bam_index
+    String transcriptome_duplicate_metrics
+    String output_bam
+    String output_bam_index
+    String duplicate_metrics
+    String rnaseqc2_gene_tpm
+    String rnaseqc2_gene_counts
+    String rnaseqc2_exon_counts
+    String rnaseqc2_fragment_size_histogram
+    String rnaseqc2_metrics
+    String picard_rna_metrics
+    String picard_alignment_summary_metrics
+    String picard_insert_size_metrics
+    String picard_insert_size_histogram
+    String picard_base_distribution_by_cycle_metrics
+    String picard_base_distribution_by_cycle_pdf
+    String picard_quality_by_cycle_metrics
+    String picard_quality_by_cycle_pdf
+    String picard_quality_distribution_metrics
+    String picard_quality_distribution_pdf
+    String? picard_fingerprint_summary_metrics
+    String? picard_fingerprint_detail_metrics
+    File unified_metrics
+
+    Int cpu = 1
+    Int memory_mb = 2000
+    Int disk_size_gb = 10
+  }
+
+  String outputs_json_file_name = "outputs_to_TDR_~{output_basename}.json"
+
+  command <<<
+        python3 << CODE
+        import json
+
+        outputs_dict = {}
+
+        # write normal outputs, adjusting field names as needed for TDR schema
+        outputs_dict["transcriptome_bam"]="~{transcriptome_bam}"
+        outputs_dict["transcriptome_bam_index"]="~{transcriptome_bam_index}"
+        outputs_dict["transcriptome_duplicate_metrics_file"]="~{transcriptome_duplicate_metrics}"
+        outputs_dict["genome_bam"]="~{output_bam}"
```

Comment:
different name for output_bam is okay I guess (just noting the difference).

Reply:
yes - and this isn't the only name that changes. this task effectively renames the outputs to match what's in the TDR dataset schema. i'll add a comment to that effect in the WDL.